forge/ee/routes/gitops/index.js

```diff
@@ -0,0 +1,119 @@
+// const axios = require('axios')
+
+module.exports = async function (app) {
+    // All routes are relative to /api/v1/teams/:teamId/git
+
+    app.addHook('preHandler', async (request, reply) => {
+        if (request.params.teamId !== undefined) {
+            request.team = await app.db.models.Team.byId(request.params.teamId)
+            if (!request.team) {
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                return
+            }
+            const teamType = await request.team.getTeamType()
+            if (!teamType.getFeatureProperty('gitIntegration', false)) {
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                return // eslint-disable-line no-useless-return
+            }
+            if (!request.teamMembership) {
+                request.teamMembership = await request.session.User.getTeamMembership(request.team.id)
+            }
+        }
+        if (!request.team) {
+            reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+            return // eslint-disable-line no-useless-return
+        }
+    })
+
+    app.get('/tokens', {
+        preHandler: app.needsPermission('team:git:tokens:list')
+    }, async (request, reply) => {
+        const tokens = await app.db.models.GitToken.byTeam(request.team.id)
+        reply.send({
+            meta: {},
+            count: tokens.length,
+            tokens: app.db.views.GitToken.tokenList(tokens)
+        })
+    })
+
+    app.post('/tokens', {
+        preHandler: app.needsPermission('team:git:tokens:create')
+    }, async (request, reply) => {
+        try {
+            const body = request.body
+            if (!body.name || !body.token) {
+                reply.code(400).send({ code: 'invalid_request', error: 'Missing required fields' })
+                return
+            }
+            const token = await app.db.models.GitToken.create({
+                name: body.name,
+                token: body.token,
+                TeamId: request.team.id
+            })
+            // TODO: audit log
+            // await app.auditLog.Project.project.httpToken.created(request.session.User, null, request.project, body)
```

Comment:
I assume still outstanding adding auditlog? I don't see any frontend code for rendering them yet

Reply:
Yes - outstanding. Will see if I can get it in tomorrow morning